src/distributions/uniform.rs

```diff
@@ -843,6 +843,26 @@ impl UniformSampler for UniformChar {
     }
 }
 
+/// Note: the `String` is potentially left with excess capacity if the range 
+/// includes non ascii chars; optionally the user may call 
+/// `string.shrink_to_fit()` afterwards.
+#[cfg(feature = "alloc")]
+impl super::DistString for Uniform<char>{
+    fn append_string<R: Rng + ?Sized>(&self, rng: &mut R, string: &mut alloc::string::String, len: usize) {
+        // Getting the hi value to assume the required length to reserve in string.
+        let mut hi = self.0.sampler.low + self.0.sampler.range;
+        if hi >= CHAR_SURROGATE_START {
+            hi += CHAR_SURROGATE_LEN;
+        }
```

Comment:
Largest possible result is one less than this I think?

Reply:
It would be nice to have some tests for such corner cases.

Reply:
I have fix this and add some test.